draft-ietf-quic-transport.md

```diff
@@ -3874,145 +3874,136 @@ The details of loss detection and congestion control are described in
+perform ECN validation using the reported ECT(1) counts.
 
 
-#### Receiving ACK Frames {#ecn-ack}
+#### Receiving ACK Frames with ECN Counts {#ecn-ack}
 
-Erroneous application of ECN marks in the network can result in degraded
-connection performance.  An endpoint that receives an ACK frame with ECN
-counts therefore validates the counts before using them. It performs this
-validation by comparing newly received counts against those from the last
-successfully processed ACK frame. Any increase in ECN counts is validated
-based on the markings that were applied to packets that are newly
-acknowledged in the ACK frame.
+Erroneous application of ECN-CE markings by the network can result in degraded
+connection performance.  An endpoint that receives an ACK frame with ECN counts
+therefore validates the counts before using them. It performs this validation by
+comparing newly received counts against those from the last successfully
+processed ACK frame. Any increase in the ECN counts is validated based on the
+ECN markings that were applied to packets that are newly acknowledged in the ACK
+frame.
 
-If an ACK frame newly acknowledges a packet that the endpoint sent with either
-ECT(0) or ECT(1) codepoints set, ECN validation fails if ECN counts are not
-present in the ACK frame.  This check detects a network element that zeroes out
-ECN bits or a peer that is unable to access ECN markings.
+If an ACK frame newly acknowledges a packet that the endpoint sent setting
+either the ECT(0) or ECT(1) codepoint, ECN validation fails if the
+corresponding ECN counts are not present in the ACK frame. This check detects a
+network element that zeroes the ECN field or a peer that is unable to access
+ECN the markings.
 
-ECN validation fails if the sum of the increase in ECT(0) and ECN-CE counts is
-less than the number of newly acknowledged packets that were originally sent
+ECN validation also fails if the sum of the increase in ECT(0) and ECN-CE counts
+is less than the number of newly acknowledged packets that were originally sent
 with an ECT(0) marking.  Similarly, ECN validation fails if the sum of the
 increases to ECT(1) and ECN-CE counts is less than the number of newly
 acknowledged packets sent with an ECT(1) marking.  These checks can detect
-removal of ECN markings in the network.
+remarking of ECN-CE markings by the network.
 
 An endpoint could miss acknowledgements for a packet when ACK frames are lost.
 It is therefore possible for the total increase in ECT(0), ECT(1), and ECN-CE
-counts to be greater than the number of packets acknowledged in an ACK frame.
-This is why counts are permitted to be larger than might be accounted for by
-newly acknowledged packets.
+counts to be greater than the number of packets that are newly acknowledged by
+an ACK frame. This is why ECN counts are permitted to be larger than the value
+corresponding to the largest acknowledged packet number.
```

Comment:
This is unfortunately incorrect.  One could in theory start with a packet number of 1 million and it should not affect correctness.

```suggestion
an ACK frame. This is why ECN counts are permitted to be larger than the total
number of acknowledged packets within a packet number space.
```

Reply:
How about:

```suggestion
an ACK frame. This is why ECN counts are permitted to be larger than the total
number of packets that are acknowledged.
```
I don't think we need to say "packet number space" again here.